src/datatype.rs

```diff
@@ -407,6 +403,73 @@ impl<'a> From<DataTypeRef<'a>> for DataType {
     }
 }
 
+/// Excel datetime type. Possible: date, time, datetime, duration.
+/// At this time we can only determine datetime (date and time are datetime too) and duration.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum ExcelDateTimeType {
+    /// DateTime
+    DateTime,
+    /// TimeDelta (Duration)
+    TimeDelta,
+}
+
+/// Structure for Excel date and time representation.
+#[derive(Debug, Clone, PartialEq)]
+pub struct ExcelDateTime {
+    value: f64,
+    datetime_type: ExcelDateTimeType,
+    is_1904: bool,
+}
+
+impl ExcelDateTime {
+    /// Creates a new `ExcelDateTime`
+    pub fn new(value: f64, datetime_type: ExcelDateTimeType, is_1904: bool) -> Self {
+        ExcelDateTime {
+            value,
+            datetime_type,
+            is_1904,
+        }
+    }
+
+    /// Converting data type into a float
+    pub fn as_f64(&self) -> f64 {
+        self.value
+    }
+
+    /// Try converting data type into a duration
+    #[cfg(feature = "dates")]
+    pub fn as_duration(&self) -> Option<chrono::Duration> {
+        let ms = self.value * MS_MULTIPLIER;
+        Some(chrono::Duration::milliseconds(ms.round() as i64))
+    }
+
+    /// Try converting data type into a datetime
+    #[cfg(feature = "dates")]
+    pub fn as_datetime(&self) -> Option<chrono::NaiveDateTime> {
+        let excel_epoch = EXCEL_EPOCH.get_or_init(|| {
+            chrono::NaiveDate::from_ymd_opt(1899, 12, 30)
+                .unwrap()
+                .and_hms_opt(0, 0, 0)
```

Comment:
I know this is just a copy paste, but `and_time(NaiveTime::MIN)` doesn't require `unwrap`. Not sure if it is more performant (cannot fail) or it is optimized out, but seems better?